Index: apps/api/.env
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/apps/api/.env b/apps/api/.env
--- a/apps/api/.env	
+++ b/apps/api/.env	
@@ -1,3 +1,3 @@
-DATABASE_URL="postgresql://postgres:workoutloud@localhost:5432/workoutloud?schema=public"
+DATABASE_URL="postgresql://postgres:workoutloud@localhost:5433/workoutloud?schema=public"
 PUBLIC_GOOGLE_API_TOKEN= "773793174033-mva0e49hdl20f7aqbcpc7h5pe6fipaej.apps.googleusercontent.com"
 SECRET= "workoutloud_secret"
\ No newline at end of file
